### DISCUSSION TITLE
There are 0 listed files in folder: test/

### DISCUSSION THREAD

**Discussion by imadgohar** [obss/sahi#678] [Q&A] [answered]
why I am getting 0 listed files in folder:
The command I am trying for inference is as under:

!sahi predict --slice_width 640 --slice_height 640 --overlap_height_ratio 0.1 --overlap_width_ratio 0.1 --source ../../dataset/images/test --model_path runs/train/exp/weights/best.pt --model_type yolov5

**Comment by fcakyon** [comment]
Have you tried providing an absolute path instead of a relative?

**Reply by imadgohar** [reply]
Yes.
I also tried the colab tutorial YOLOV5 + SAHI, same issue there.

**Reply by imadgohar** [reply]
@fcakyon 
Do we need to pass .JSON annotation file of the test data during predict or just source and trained model is enough as I did above.

**Reply by imadgohar** [reply]
Here is the complete code on colab I am running:

!pip install -U torch sahi yolov5

#from sahi import AutoDetectionModel
from sahi.model import Yolov5DetectionModel
from sahi.utils.cv import read_image
from sahi.utils.file import download_from_url
from sahi.predict import get_prediction, get_sliced_prediction, predict
from IPython.display import Image

yolov5_model_path = '/content/drive/MyDrive/SAHI/yolov5-SAHI/best.pt'
#yolov5_model_path = 'best.pt'

model_type = "yolov5"
model_path = yolov5_model_path
model_device = 'cuda:0'
#model_confidence_threshold = 0.4

slice_height = 640
slice_width = 640
overlap_height_ratio = 0.1
overlap_width_ratio = 0.1

source_image_dir = "/content/drive/MyDrive/SAHI/yolov5-SAHI/test/"
#source_image_dir = "test/"

predict(
    model_type=model_type,
    model_path=model_path,
    model_device=model_device,
    #model_confidence_threshold=model_confidence_threshold,
    source=source_image_dir,
    slice_height=slice_height,
    slice_width=slice_width,
    overlap_height_ratio=overlap_height_ratio,
    overlap_width_ratio=overlap_width_ratio,
)

This is the outpu:
There are 0 listed files in folder: test/
Performing inference on images: 0it [00:00, ?it/s]Prediction results are successfully exported to runs/predict/exp4

!sahi version
0.10.7

**Reply by fcakyon** [reply]
No need for json when doing prediction. What you do should be enough. What are the file extensions in the source folder?

**Reply by imadgohar** [reply]
It's .JPG in capital

**Reply by fcakyon** [reply]
can you test it after converting some of the extensions into lower letters (.jpg). If the issue is solved, then it means there is a bug in SAHI that prevents the code to find uppercase extension files.

**Reply by imadgohar** [reply]
Thanks @fcakyon for your quick responses, Yes it works now the problem is with the image extension. SAHI does not accept the extension in uppercase.

**Reply by fcakyon** [reply]
@imadgohar this pr should fix this: https://github.com/obss/sahi/pull/713

**Comment by fcakyon** [comment, marked as answer]
> Thanks @fcakyon for your quick responses, Yes it works now the problem is with the image extension. SAHI does not accept the extension in uppercase.

this pr adds support to extensions in uppercase: https://github.com/obss/sahi/pull/713
